packages/tools/src/utilities/segmentation/createLabelmapMemo.ts

```diff
@@ -47,6 +46,18 @@ export function restoreMemo(isUndo?: boolean) {
   });
   const slices = useVoxelManager.getArrayOfModifiedSlices();
   triggerSegmentationDataModified(this.segmentationId, slices);
+
+  // Trigger the HISTORY_UNDO event
+  eventTarget.dispatchEvent(
+    new CustomEvent(Events.HISTORY_UNDO, {
```

Comment:
This should probably fire generically on the history undo/redo after the restoreMemo call.  Otherwise it can't be used for listening to other types of undo/redo changes.  It should also include the memo itself rather than including the segmentationId so that it is a generic event type.

Reply:
If you want to leave it here for now, moving it later, at least add the memo object itself rather than the segmentationId so that it has the right object design to work.